packages/core/src/__tests__/acceptance/application.shutdown.acceptance.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect, skipIf} from '@loopback/testlab';
+import {ChildProcess, fork} from 'child_process';
+
+const app = require.resolve('./application-with-shutdown');
+const isWindows = process.platform === 'win32';
+
+describe('Application shutdown hooks', () => {
+  skipIf(isWindows, it, 'traps registered signals - SIGTERM', () => {
+    return testSignal('SIGTERM');
+  });
+
+  skipIf(
+    isWindows,
+    it,
+    'traps registered signals with grace period - SIGTERM',
+    () => {
+      // No 'stopped` is recorded
+      return testSignal('SIGTERM', 5, ['start\n', 'stop\n']);
+    },
+  );
+
+  skipIf(isWindows, it, 'traps registered signals - SIGINT', () => {
+    return testSignal('SIGINT');
+  });
+
+  skipIf(isWindows, it, 'does not trap unregistered signals - SIGHUP', () => {
+    return testSignal('SIGHUP', undefined, ['start\n']);
+  });
+
+  function normalizeStdoutData(output: string[]) {
+    // The received events can be `['start\n', 'stop\nstopped\n']` instead
```

Comment:
What is the difference in these two lists in the comments?

Reply:
The array has different number of items.

Reply:
> The array has different number of items.

array 1: ['start\n', 'stop\nstopped\n']        has two items: 'start\n' and 'stop\nstopped'
array2:   [ 'start\n', 'stop\n', 'stopped\n' ] has three items: 'start'\n, 'stop\n', 'stopped\n'

why does the second item in the first array contain 2 items separated by \n ?  Is this a typo?

Reply:
No, that's how Node.js 8.x writes data to the stream - it ends up with one `data` event instead of two (in Node 10.x and above).